server/prompt_improvement_skill.py

```diff
@@ -0,0 +1,64 @@
+from pydantic import BaseModel, field_validator
+from adala.skills import Skill
+from typing import Any, Dict, List, Tuple
+from adala.skills.collection.text_generation import TextGenerationSkill
+
+
+class PromptImprovementSkillResponseModel(BaseModel):
+    
+    # hidden variable, used for validation
+    _input_variables: List[str]
+    reasoning: str
+    improved_user_prompt: str
+    # NOTE: not exposed in LSE yet, so default is always used. Should improve this as well when we expose it.
+    # improved_system_prompt: str
+
+
+    @field_validator("improved_user_prompt", mode="after")
+    def validate_used_variables(cls, value: str) -> str:
+
+        start_variable_idx = value.find("{")
```

Comment:
annoyingly, I think we're probably best off using a regex for this kind of thing. We'd have false positives using this for e.g. "blah {blah blah} blah"

Reply:
agree, we have a special function for this case: https://github.com/HumanSignal/Adala/blob/master/adala/utils/parse.py#L39

Reply:
done